tests/into_iterator.rs

```diff
@@ -45,3 +45,25 @@ impl ::core::iter::IntoIterator for Numbers3 {
         <Vec<i32> as ::core::iter::IntoIterator>::into_iter(self.numbers)
     }
 }
+
+#[derive(IntoIterator)]
+struct Generic1<T> {
+    #[into_iterator(owned, ref, ref_mut)]
+    items: Vec<T>,
+}
+
+#[derive(IntoIterator)]
+struct Generic2<'a, T, U: Send>
+where
+    T: Send,
+{
+    #[into_iterator(owned, ref, ref_mut)]
+    items: Vec<T>,
+    useless: &'a U,
+}
+
+#[derive(IntoIterator)]
+struct Generic3<'a, 'b, T> {
+    #[into_iterator(owned, ref, ref_mut)]
+    items: &'a mut Vec<&'b mut T>,
```

Comment:
@MegaBluejay good 👍

It's also worth to add the tests which check just `#[into_iterator]` attribute (without `owned`/`ref` inside) against generics.

Reply:
> It's also worth to add the tests which check just `#[into_iterator]` attribute (without `owned`/`ref` inside) against generics.

Done

Reply:
@MegaBluejay yup, this is out-of-scope. So, recompose the test the way it makes sense.